src/sciline/visualize.py

```diff
@@ -85,47 +93,45 @@ def get_subgraph_name(name: str) -> str:
         return sgname
 
     subgraphs: Dict[str, FormattedGraph] = {}
-    for p, (p_name, args, ret) in graph.items():
-        subgraph_name = get_subgraph_name(ret.name)
+    for p, formatted_p in graph.items():
+        subgraph_name = get_subgraph_name(formatted_p.ret.name)
         if subgraph_name not in subgraphs:
             subgraphs[subgraph_name] = {}
-        subgraphs[subgraph_name][p] = (p_name, args, ret)
+        subgraphs[subgraph_name][p] = formatted_p
     return subgraphs
 
 
 def _add_subgraph(graph: FormattedGraph, dot: Digraph, subgraph: Digraph) -> None:
-    for p, (p_name, args, ret) in graph.items():
+    for p, formatted_p in graph.items():
         unsatisfied = f'{_qualname(HandleAsComputeTimeException.handle_unsatisfied_requirement)}.<locals>.unsatisfied_sentinel'  # noqa: E501
-        if p_name == unsatisfied:
+        if formatted_p.name == unsatisfied:
             subgraph.node(
-                ret.name,
-                ret.name,
-                shape='box3d' if ret.collapsed else 'rectangle',
+                formatted_p.ret.name,
+                formatted_p.ret.name,
+                shape='box3d' if formatted_p.ret.collapsed else 'rectangle',
                 color='red',
                 fontcolor='red',  # Set text color to red
                 style='dashed',
             )
         else:
             subgraph.node(
-                ret.name, ret.name, shape='box3d' if ret.collapsed else 'rectangle'
+                formatted_p.ret.name,
+                formatted_p.ret.name,
+                shape='box3d' if formatted_p.ret.collapsed else 'rectangle',
             )
         # Do not draw dummy providers created by Pipeline when setting instances
-        if p_name in (
-            f'{_qualname(Pipeline.__setitem__)}.<locals>.<lambda>',
-            f'{_qualname(Pipeline.set_param_table)}.<locals>.<lambda>',
-            unsatisfied,
-        ):
+        if formatted_p.kind in ('parameter', 'table', 'sentinel'):
```

Comment:
What is `sentinel` here? And why are we still using `unsatisfied` above but not here?